### ISSUE TITLE
`if(false)` elimination at `-O0`

### ISSUE THREAD

**Issue by JohanEngelen** [ldc-developers/ldc#2344]
Should we do dead code elimination on `if (false)` at `-O0`?
GDC and DMD do so.

```d
void foo() {
	if (false) {
		int a = 1;	
	}
}
```
gives (LDC 1.4.0)
```asm
void example.foo():
        xor     eax, eax
        mov     cl, al
        test    cl, 1
        jne     .LBB0_1
        jmp     .LBB0_2
.LBB0_1:
        mov     dword ptr [rsp - 4], 1
.LBB0_2:
        ret
```

**Comment by kinke** [comment]
Good catch; clang does it too, so I think we should.

**Comment by JohanEngelen** [comment]
Similar work can be done for `if(true){}else{}`, `while(false)`.

Note that this is more complicated than it may appear.
```d
void foo(int a) {
    int temp;
    if (a) goto lab1;
    if (false) {
        lab1:
        temp = 42;
    }
}
```

**Comment by baziotis** [comment]
Probably we can now close this issue :)

**Comment by JohanEngelen** [comment]
yep, implemented by #3134
